app/src/main/java/com/example/tasky/agenda/presentation/agenda_detail/AgendaDetailViewModel.kt

```diff
@@ -141,19 +162,58 @@ class AgendaDetailViewModel(
                             }
                         }
                     }
-                _state.update { it.copy(imageLoading = false) }
+
+                    updateDetails<AgendaItemDetails.Event> { event ->
+                        event.copy(isImageLoading = false)
+                    }
                 }
             }
             is AgendaDetailAction.OnPhotoDelete -> {
-                _state.update { it ->
-                    it.detailsAsEvent()?.let { eventDetails ->
-                        val updatedDetails = eventDetails.copy(
-                            photos = eventDetails.photos.filterNot { photo ->
-                                photo.id == action.photoId
-                            }
+                updateDetails<AgendaItemDetails.Event> { event ->
+                    event.copy(
+                        photos = event.photos.filterNot { photo ->
+                            photo.id == action.photoId
+                        }
+                    )
+                }
+            }
+            AgendaDetailAction.OnAddAttendeeClick -> {
+                _state.update { it.copy(agendaDetailBottomSheetType = AgendaDetailBottomSheetType.ADD_ATTENDEE) }
+            }
+            AgendaDetailAction.OnDeleteAgendaItemClick -> {
+                _state.update { it.copy(agendaDetailBottomSheetType = AgendaDetailBottomSheetType.DELETE_AGENDA_ITEM) }
+            }
+            AgendaDetailAction.OnDismissBottomSheet -> {
+                _state.update { it.copy(agendaDetailBottomSheetType = AgendaDetailBottomSheetType.NONE) }
+            }
+            is AgendaDetailAction.OnAttendeeEmailValueChanged -> {
+                updateDetails<AgendaItemDetails.Event> { event ->
+                    event.copy(
+                        attendeeEmail = action.email
+                    )
+                }
+            }
+            is AgendaDetailAction.OnAttendeeEmailFieldFocusChanged -> {
```

Comment:
I think the way it’s wired right now makes the error feel a bit too sensitive, since it fires immediately on focus change. What I’d suggest instead is a slightly softer flow: when the field gains focus, clear out any existing error so the user can type without red underlines hanging around. While they’re typing, just keep the value live and don’t validate yet. Then, once they leave the field, that’s the moment to check the email if it’s not empty and surface an error if it’s invalid. Finally, when they actually hit “add” or submit, that’s where you do the strict validation and either commit the attendee or block with an error.